pkg/blockbuilder/blockbuilder.go

```diff
@@ -203,7 +203,7 @@ func (b *BlockBuilder) nextConsumeCycle(ctx context.Context, cycleEndTime time.T
 			continue
 		}
 
-		state := partitionStateFromLag(b.logger, lag, b.fallbackOffsetMillis)
+		state := PartitionStateFromLag(b.logger, lag, b.fallbackOffsetMillis)
```

Comment:
Note: I will follow up with moving many of these Kafka utilities to the scheduler package. Long-term, scheduler will be their sole user.